### PR TITLE
OHM-543: updated rest service url to include path

### PR DESCRIPTION
**Pull request by nthfloor2**
Added new config parameter and updated rest service to check for this path and append it to the server url.

### CODE REVIEW

**Comment by nthfloor2** [comment]
Had to update config property for each test file - we should rethink this approach to rather pull config settings from a central location instead of defining it for each file.

File: app/config/default.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "minimumCoreVersion": "4.0.0-rc.4",
   "protocol": "https",
   "host": "localhost",
+  "host_path": "",
```
Comment (bausmeier) [inline app/config/default.json]: @nthfloor2 All of the other keys are named using camel case. We should stick to that convention for this one as well.

File: app/scripts/services/rest.js [hunk 2]
```diff
@@ -2,8 +2,9 @@ export function Api ($rootScope, $resource, config) {
   // fetch API server details
   let protocol = config.protocol
```
Comment (andrew-w-ross) [inline app/scripts/services/rest.js]: Change let to const.

File: app/scripts/services/rest.js [hunk 3]
```diff
@@ -2,8 +2,9 @@ export function Api ($rootScope, $resource, config) {
...
   let host = config.host
+  let hostPath = config.hostPath.startsWith('/') ? config.hostPath.replace(/\/$/, '') : '/' + config.hostPath.replace(/\/$/, '')
```
Comment (andrew-w-ross) [inline app/scripts/services/rest.js]: Can we just use a single regex for this?

Comment (nthfloor2) [reply app/scripts/services/rest.js]: @andrew-w-ross not sure how you would do this using regex, but there is a nifty string library which can do this check and replace, so have included and used it instead.

**Pull request merged**